service/domain/feeds/content/transport/mapping_contact.go

```diff
@@ -46,45 +48,84 @@ var contactMapping = MessageContentMapping{
 	},
 }
 
-func unmarshalContactAction(t transportContact) (content.ContactAction, error) {
-	if t.Following && !t.Blocking {
-		return content.ContactActionFollow, nil
-	}
+func unmarshalContactAction(t transportContact) (content.ContactActions, error) {
+	var actions []content.ContactAction
 
-	if !t.Following && !t.Blocking {
-		return content.ContactActionUnfollow, nil
+	if v := t.Following; v != nil {
+		if *v {
+			actions = append(actions, content.ContactActionFollow)
+		} else {
+			actions = append(actions, content.ContactActionUnfollow)
+		}
 	}
 
-	if !t.Following && t.Blocking {
-		return content.ContactActionBlock, nil
+	if v := t.Blocking; v != nil {
+		if *v {
+			actions = append(actions, content.ContactActionBlock)
+		} else {
+			actions = append(actions, content.ContactActionUnblock)
+		}
 	}
 
-	return content.ContactAction{}, errors.New("invalid contact action")
+	return content.NewContactActions(actions)
 }
 
-func marshalContactAction(action content.ContactAction, t *transportContact) error {
-	switch action {
-	case content.ContactActionFollow:
-		t.Following = true
-		t.Blocking = false
-		return nil
-	case content.ContactActionUnfollow:
-		t.Following = false
-		t.Blocking = false
-		return nil
-	case content.ContactActionBlock:
-		t.Following = false
-		t.Blocking = true
-		return nil
-	default:
-		return fmt.Errorf("unknown contact action '%#v'", action)
+func marshalContactActions(actions content.ContactActions, t *transportContact) error {
+	for _, action := range actions.List() {
+		switch action {
+		case content.ContactActionFollow:
+			t.Following = internal.Ptr(true)
+		case content.ContactActionUnfollow:
+			t.Following = internal.Ptr(false)
+		case content.ContactActionBlock:
+			t.Blocking = internal.Ptr(true)
+		case content.ContactActionUnblock:
+			t.Blocking = internal.Ptr(false)
+		default:
+			return fmt.Errorf("unknown contact action '%#v'", action)
+		}
 	}
+
+	return nil
 }
 
 type transportContact struct {
-	messageContentType        // todo this is stupid
-	Contact            string `json:"contact"`
-	Following          bool   `json:"following"`
-	Blocking           bool   `json:"blocking"`
+	messageContentType // todo this is stupid
+	Contact            string
+	Following          *bool
+	Blocking           *bool
 	// todo pub field
 }
+
+// MarshalJSON implements custom logic which fixes the default behaviour of the
+// encoding/json package which treats nil pointers and pointers to zero values
+// in the same way when omitempty is specified. For example normally both a bool
+// pointer which is nil and a bool pointer which is set to false will not appear
+// in resulting JSON when omitempty is used.
```

Comment:
Are you sure it's the case? I'm pretty sure that while using `omitempty` with a bool pointer you should get the expected behavior which is:
1) nil => no entry in JSON 
2) pointer to true => "true" in JSON 
3) pointer to false => "false" in JSON 

Here's a snippet to prove it: https://goplay.tools/snippet/gFvsSh6DgyV

Reply:
I see that now but I would swear that my tests were failing previously, it is possible that this was due to some other bug.

Reply:
[fa70362](https://github.com/planetary-social/scuttlego/pull/58/commits/fa70362fce97c67bc56fa4875c0812b4f35b2f10)